Publish compiled files and type definition files to NPM

micheal-parks opened the pull request.

micheal-parks commented on web/frontend/package.json:
```diff
@@ -2,52 +2,58 @@
+      "import": "./dist/rc.js"
+    }
+  },
   "dependencies": {
```
Dependencies / DevDependencies have been reorganized according to what's shared with app. This will ultimately result in node_modules for app being smaller due to unnecessary packages not being downloaded.

micheal-parks commented on web/frontend/scripts/copy-prime-assets.js:
```diff
@@ -1,6 +1,5 @@
 /* eslint-disable @typescript-eslint/no-var-requires */
-/* eslint-disable unicorn/prefer-module */
-const fs = require('node:fs');
+import fs from 'node:fs';
```
I've aligned the package.json for RDK to use `type: module` like our other repos.

micheal-parks commented on web/frontend/src/main-lib.ts:
```diff
@@ -0,0 +1,18 @@
+import type { Credentials } from '@viamrobotics/rpc';
+import type { Client } from '@viamrobotics/sdk';
+
+export const createRcApp = (props: {
```
This is what'll be exported to NPM. App will do this:
```ts
const app = createRcApp({
  host,
  bakedAuth,
  supportedAuthTypes,
  webrtcEnabled,
  client,
})
app.mount('#rc-container')

// later
app.unmount()

michaellee1019 reviewed (approved):
LGTM. Thanks for doings this. I spent some time deploying an AppImage to a robot, everything is fine. Just wanted to make sure that everything is still packaged correctly with these changes.

edaniels commented:
Was this in response to something that broke?

micheal-parks commented:
@michaellee1019 I've fixed all TS issues and removed all @ts-nocheck statements. I don't think you need to do another testing round, no logic was really changed by the fixes.
